modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeStaticValueResolver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.resolver;
+
+import static org.mule.runtime.core.util.ClassUtils.isInstance;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.MuleContext;
+
+/**
+ * {@link ValueResolver} implementation which always returns the same value, but if it is required,
+ * the value will be converted to the required type using the mule transformers.
+ *
+ * @param <T> Type of the value to resolver
+ * @since 4.0
+ */
+public class TypeSafeStaticValueResolver<T> implements ValueResolver<T> {
```

Comment:
This approach seems overkill.. What is the advantage of having a `TypeSafeStaticValueResolver`, plus a `TypeSafeExpressionValueResolver` which both compose a `TypeSafeTransformer` (which is not actually a transformer) vs simply having a `TypeSafeValueResolverDecorator` which can  work with any kind of ValueResolver we may need.